tests/storage-spec.js

```diff
@@ -0,0 +1,179 @@
+import '../app/scripts/src/main';
+
+describe('languageStorage factory', function () {
+
+    beforeAll(function () {
+
+        this.setupModule = function () {
+
+            angular.mock.module('language-select', ($provide) => {
+                $provide.service('languageSelectConfig', () => {
+                    return {
+                        availableLanguages: () => this.languageChoices,
+                        defaultLanguageId: () => 'se',
+                    };
+                });
+            });
+
+            inject(function (languageStorage, languageSelectConfig, $cookies, $rootScope) {
+                this.languageStorage = languageStorage;
+                this.languageSelectConfig = languageSelectConfig;
+                this.$cookies = $cookies;
+                this.$rootScope = $rootScope;
+            });
+        };
+
+    });
+
+    describe('methods', function () {
+
+        beforeEach(function () {
+
+            this.languageChoices = [
+                {
+                    id: 'en_gb',
+                    label: 'English',
+                },
+                {
+                    id: 'pl',
+                    label: 'Polski',
+                },
+                {
+                    id: 'se',
+                    label: 'Svensk',
+                },
+            ];
+            this.setupModule();
+
+            spyOn(this.$rootScope, '$broadcast');
+        });
+
+        describe('getLanguageChoices method', function () {
+
+            it('should return the language choices from the config', function () {
+                const choices = this.languageStorage.getLanguageChoices();
+                expect(choices).toBe(this.languageChoices);
```

Comment:
Similar to previous comment: `toEqual` may be more stable if object reference isn't required by this method.